### PR TITLE
RBAC implementation for applets

### PR DESCRIPTION
**Pull request by mikeyroy26**

### CODE REVIEW

File: api/src/app/oidc.py [hunk 1]
```diff
@@ -29,4 +29,9 @@ def get_current_user_groups():
 def get_current_user_roles():
     token = request.headers['authorization'].split(' ')[1]
     user_info = oidc._get_token_info(token)
-    return user_info['realm_access']['roles']
\ No newline at end of file
+    return user_info['realm_access']['roles']
```
Comment (CodeCrafter7) [inline api/src/app/oidc.py:32]: @mikeyroy26  , please use the new get_user_id method, you can take a pull from the main branch

Comment (mikeyroy26) [reply api/src/app/oidc.py:32]: ok @CodeCrafter7

File: api/src/app/service/userdata_service.py
Comment (CodeCrafter7) [inline api/src/app/service/userdata_service.py:1]: @mikeyroy26 please update the chain entity with the proper columns

Comment (mikeyroy26) [reply api/src/app/service/userdata_service.py:1]: @CodeCrafter7 as of now acl config is used in query to fetch the records and result is not binding to entity/dto.
Entity changes can be done later.

**Review by CodeCrafter7 — APPROVED** [review]

**Pull request merged**